### PR TITLE
octopus: rgw: check object locks in multi-object delete 

### PR DESCRIPTION
**Pull request by trociny**
backport tracker: https://tracker.ceph.com/issues/48424

---

backport of https://github.com/ceph/ceph/pull/37933
parent tracker: https://tracker.ceph.com/issues/47586

this backport was staged using ceph-backport.sh version 16.0.0.6848
find the latest version at https://github.com/ceph/ceph/blob/master/src/script/ceph-backport.sh

### CODE REVIEW

**Comment by trociny** [comment]
d8c20265c1329bd61b799fc226333f2f09acdd2e has not been cherry-picked as not needed.

File: src/rgw/rgw_op.cc [hunk 1]
```diff
@@ -6483,9 +6488,12 @@ int RGWDeleteMultiObj::verify_permission()
         }
       }
     }
+
+    bool empty = s->object.empty() || s->object.instance.empty();
+
     auto usr_policy_res = eval_user_policies(s->iam_user_policies, s->env,
                                               boost::none,
-                                              s->object.instance.empty() ?
+                                              empty ?
```
Comment (mattbenjamin) [inline src/rgw/rgw_op.cc]: not clear why (iirc) this is lexically different from the change on master, but, ok

File: src/rgw/rgw_op.cc [hunk 2]
```diff
@@ -6489,11 +6489,11 @@ int RGWDeleteMultiObj::verify_permission()
       }
     }
 
-    bool empty = s->object.empty() || s->object.instance.empty();
+    bool not_versioned = s->object.empty() || s->object.instance.empty();
```
Comment (mattbenjamin) [inline src/rgw/rgw_op.cc:6492]: ok, that was fixed by mark

**Review by mattbenjamin — APPROVED** [review]
lgtm

**Comment by mattbenjamin** [comment]
@trociny can you confirm, are you working on the nautilus backport, as well?

**Comment by trociny** [comment]
I have plans to push the nautilus backport as well, though I thought it would be easier to wait for this backport to be merged so I could cherry-pick these commits instead of resolving the master conflicts again.

**Comment by trociny** [comment]
Ok. I see now you pushed the nautilus PR #41164

**Comment by smithfarm** [comment]
oops! Looks like we missed the 15.2.13 boat here.

**Pull request merged**